MediaBrowser.Api/Syncplay/SyncplayService.cs

```
@@ -0,0 +1,267 @@
+    {
+        /// <summary>
+        /// The session manager.
+        /// </summary>
+        private readonly ISessionManager _sessionManager;
+
+        /// <summary>
+        /// The session context.
+        /// </summary>
+        private readonly ISessionContext _sessionContext;
+
+        /// <summary>
+        /// The Syncplay manager.
+        /// </summary>
+        private readonly ISyncplayManager _syncplayManager;
+
+        public SyncplayService(
+            ILogger<SyncplayService> logger,
+            IServerConfigurationManager serverConfigurationManager,
+            IHttpResultFactory httpResultFactory,
+            ISessionManager sessionManager,
+            ISessionContext sessionContext,
+            ISyncplayManager syncplayManager)
+            : base(logger, serverConfigurationManager, httpResultFactory)
+        {
+            _sessionManager = sessionManager;
+            _sessionContext = sessionContext;
+            _syncplayManager = syncplayManager;
+        }
+
+        /// <summary>
+        /// Handles the specified request.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        public void Post(SyncplayNewGroup request)
+        {
+            var currentSession = GetSession(_sessionContext);
+            _syncplayManager.NewGroup(currentSession);
+        }
+
+        /// <summary>
+        /// Handles the specified request.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        public void Post(SyncplayJoinGroup request)
+        {
+            var currentSession = GetSession(_sessionContext);
+            var joinRequest = new JoinGroupRequest();
+            joinRequest.GroupId = Guid.Parse(request.GroupId);
+            try
+            {
+                joinRequest.PlayingItemId = Guid.Parse(request.PlayingItemId);
+            }
+            catch (ArgumentNullException)
+            {
+                // Do nothing
```

can we at least log warnings or something here?

---

The "specification" I had in mind was that both `null` and empty strings would be considered by the server to be valid. It is fine for a client to request to join a group without sending a value for `PlayingItemId`: it means that the client isn't playing anything.

Should I pick an option and consider the other as invalid and thus log it?

---

This (what you wrote in an answer) is an excellent comment, much better than `// Do nothing` in the code currently :)
Please replace this `// Do nothing` with the real explanation of why we're ignoring the exception.

Or maybe even better check explicitly, like
```cs
if (!String.IsNullOrEmpty(request.PlayingItemId))
{
    try
    {
        joinRequest.PlayingItemId = Guid.Parse(request.PlayingItemId);
    }
    catch (... whatever ...)
    {
        // log error, quit method - this is clearly a malformed request
    }
}